xpdacq/calib.py

```diff
@@ -78,6 +79,117 @@ def run_calibration(exposure=60, calibrant_file=None, wavelength=None,
     gaussian : int, optional
         gaussian width between rings, Default is 100.
     """
+
+    _check_obj(_REQUIRED_OBJ_LIST)
+    ips = get_ipython()
+    bto = ips.ns_table['user_global']['bt']
+    prun = ips.ns_table['user_global']['prun']
+
+    # d-spacing
+    if calibrant_file is not None:
+        calibrant_name = os.path.split(calibrant_file)[1]
+        calibrant_name = os.path.splitext(calibrant_name)[0]
+    else:
+        calibrant_name = 'Ni'
+
+    # scan
+    calib_collection_uid = str(uuid.uuid4())
+    calibration_dict = {'sample_name':calibrant_name,
+                        'sample_composition':{calibrant_name :1},
+                        'is_calibration': True,
+                        'calibration_collection_uid': calib_collection_uid}
+    sa = Sample(bto, calibration_dict)
+    prun_uid = prun(calibration_dict, ScanPlan(bto, ct, exposure))
+    light_header = glbl.db[-1]
+    if dark_sub:
+        dark_uid = light_header.start['sc_dk_field_uid']
+        dark_header = glbl.db[dark_uid]
+        dark_img = np.asarray(glbl.db.get_images(dark_header,
+                                glbl.det_image_field)).squeeze()
+    for ev in glbl.db.get_events(light_header, fill=True):
+        img = ev['data'][glbl.det_image_field]
+        if dark_sub:
+            img -= dark_img
+
+    print('{:=^20}'.format("INFO: you are able to calib, please refer"
+                           "to guid here:\n"))
+    print('{:^20}'
+          .format("http://xpdacq.github.io/usb_Running.html#calib-manual"))
+    print()
+    # calibration, return a azimuthal integrator
+    ai = calibration(img, calibrant_file=calibrant_file,
+                     calib_collection_uid=calib_collection_uid,
+                     wavelength=wavelength, detector=detector,
+                     gaussian=gaussian)
+
+    # masking
+    print('INFO: create mask')
```

Comment:
We should be careful here.  The correct workflow is that the mask is created "during the calibration workflow".  There are two possibilities:
1) The automask from nickel (or whatever calibrant), plus the edge mask, plus (maybe) the polygon mask for the beamstop mount,  is adequate for all samples.  
2) We need to collect data from a weakly scattering sample such as an empty kapton tube, run automask on that to get the auto-mask part of the mask, then create the mask as automask-mask+edge-mask(+beamstop mask).

If (1) is correct then the workflow coded here is acceptable.  If we need to use (2) then this is not acceptable.

Either way, it may be better to have a separate mask-creation logic that is run outside of `run_calibration`.  It is generally better that functions just do what they appear to do.  So `run_calibration()` does just that...it runs the get-calibration logic.  Then we have a `run_mask_builder()` function that does that....it runs the automask and builds the mask. 

In this way, we can leave it to the user what sample they want to mount to get the automask they want.   The documentation will then have to reflect this.  The sequence could be:
1) load Ni
2) `run_calibration()`
3) load empty kapton
4) `run_mask_builder()`
5) load sample
6) `prun(sampleObject,scanPlanObject)`
7)  `view_last_image()`
8)  toggle mask on and off in `XPDsuite` to make sure that the auto-built mask is acceptable
9)  all good?  start experiment.....